pandas/tests/frame/test_missing.py

```diff
@@ -983,3 +983,11 @@ def test_interp_time_inplace_axis(self, axis):
         result = expected.interpolate(axis=0, method="time")
         expected.interpolate(axis=0, method="time", inplace=True)
         tm.assert_frame_equal(result, expected)
+
+    def test_interp_string_axis(self):
+        # GH 25190
+        x = np.linspace(0, 100, 1000)
+        y = np.sin(x)
+        df = pd.DataFrame(data=np.tile(y, (10, 1)), index=np.arange(10), columns=x)
+        df.reindex(columns=x * 1.005).interpolate(method="linear", axis="columns")
+        df.reindex(columns=x * 1.005).interpolate(method="linear", axis="index")
```

Comment:
@jbrockmendel : Does it make sense to also check the results?

Reply:
i think that'd be worthwhile if it isnt too onerous.  if nothing else, could check that we get the same result as we would if we passed the ints instead of strings.

@YagoGG o do we need the corresponding tests for Series?